### PR TITLE
8251999: remove usage of PropertyResolvingWrapper in vmTestbase/nsk/aod

### PR DESCRIPTION
**Pull request by iignatev**
the patch 
- removes `PropertyResolvingWrapper` from `vmTestbase/nsk/aod` tests
- updates `AODRunnerArgParser` to remove surrounding `"` symbols from `javaOpts` option values
- updates vmTestbase/nsk/aod test descriptions to have `"` in `javaOpts` values (as opposed of around `-javaOpts=$value` as it's now)
- mechanically reformats/cleans up the tests (whitespaces, imports, etc) 

testing:
* [x] vmTestbase/nsk/aod on  {macosx,windows,linux}-x64 
* [x] vmTestbase/nsk/aod on macosx-x64 w/ `VM_OPTIONS`, `JAVA_OPTIONS` being empty and having value

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8251999](https://bugs.openjdk.java.net/browse/JDK-8251999): remove usage of PropertyResolvingWrapper in vmTestbase/nsk/aod

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)
 * [Serguei Spitsyn](https://openjdk.java.net/census#sspitsyn) (@sspitsyn - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/311/head:pull/311`
`$ git checkout pull/311`

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachine/VirtualMachine03/VirtualMachine03.java [hunk 1]
```diff
@@ -63,23 +63,21 @@ public void doTestActions(String targetVMId) throws Throwable {
...
+                TestUtils.assertEquals(vm1.id(), currentVMId, "vm.id() returns unexpected value: " + vm1.id());
+                TestUtils.assertEquals(vm2.id(), targetVMId, "vm.id() returns unexpected value: " + vm2.id());
+
+                TestUtils.assertTrue(!vm1.equals(vm2), vm1 + ".equals(" + vm2 + ") returns 'true'");
+
+                checkVM(vm1);
+                checkVM(vm2);
+            } finally {
+                vm2.detach();
+            }
```
Comment (shipilev) [inline test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachine/VirtualMachine03/VirtualMachine03.java:78]: Is nesting needed here because `vm1`/`vm2` can be `null` at `finally`? Then it is cleaner IMO to just do:

```
} finally {
    if (vm1 != null) vm1.detach();
    if (vm2 != null) vm2.detach();
}
```

File: test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachineDescriptor/VirtualMachineDescriptor01/VirtualMachineDescriptor01.java [hunk 2]
```diff
@@ -108,8 +109,7 @@ public void doTestActions(String targetVMId) throws Throwable {
                     "VirtualMachineDescriptor.equals() returns 'false' for '" + targetVMDesc + "' and '" + targetVMDesc2 + "'");
 
             TestUtils.assertEquals(targetVMDesc.hashCode(), targetVMDesc2.hashCode(),
-                    "VirtualMachineDescriptor.hashCode() returns different values " + "(" + targetVMDesc.hashCode() + " and " + targetVMDesc2.hashCode() + ")" +
-                    " for '" + targetVMDesc + "' and '" + targetVMDesc2 + "'");
+                    "VirtualMachineDescriptor.hashCode() returns different values for '" + targetVMDesc + "' and '" + targetVMDesc2 + "'");
```
Comment (shipilev) [inline test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachineDescriptor/VirtualMachineDescriptor01/VirtualMachineDescriptor01.java:112]: Does `TestUtils.assertEquals` print out actual/expected value as well? If not, dropping `hashCode` printout here loses debugging data.

**Review by shipilev — APPROVED** [review]
Looks good, modulo minor comments below.

**Review by sspitsyn — APPROVED** [review]
Igor, the fix looks good to me.
Thank you for all these style and import changes!

File: test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachineDescriptor/VirtualMachineDescriptor01/VirtualMachineDescriptor01.java (hunk 2 above)
Comment (iignatev) [reply test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachineDescriptor/VirtualMachineDescriptor01/VirtualMachineDescriptor01.java:112]: it does for both a reference type and `int`, [TestUtils.java]:
```
    public static void assertEquals(int i1, int i2) {
        if (i1 != i2) {
            throw new TestFailure(
                    format("Check failed: %d != %d", i1, i2));
        }
    }

    public static void assertEquals(Object obj1, Object obj2, Object errorMessage) {
        assertTrue(obj1.equals(obj2), new LazyFormatString("%s: [%s] != [%s]", errorMessage,  obj1, obj2));
    }
```
[TestUtils.java]: https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/vmTestbase/nsk/share/test/TestUtils.java#L70

File: test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachine/VirtualMachine03/VirtualMachine03.java (hunk 1 above)
Comment (iignatev) [reply test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachine/VirtualMachine03/VirtualMachine03.java:78]: nested try-finally isn't semantically equivalent with the snippet you suggested, `VirtualMachine::detach` can throw IOE, so I'd prefer to two finally blocks to guarantee that we at least to detach from both vm-s.

File: test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachineDescriptor/VirtualMachineDescriptor01/VirtualMachineDescriptor01.java (hunk 2 above)
Comment (shipilev) [reply test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachineDescriptor/VirtualMachineDescriptor01/VirtualMachineDescriptor01.java:112]: OK, cool.

File: test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachine/VirtualMachine03/VirtualMachine03.java (hunk 1 above)
Comment (shipilev) [reply test/hotspot/jtreg/vmTestbase/nsk/aod/VirtualMachine/VirtualMachine03/VirtualMachine03.java:78]: Okay then.

**Comment by iignatev** [comment]
@shipilev , @sspitsyn, thanks for your reviews.
